Xamarin.Forms.Platform.Android/Platform.cs

```diff
@@ -1134,17 +1134,23 @@ protected override int GetChildDrawingOrder(int childCount, int i)
 			{
 				//On Material design the button states use Elevation property, we need to make sure
 				//we update the elevation of other controls to be over the previous one
-				if (!_minimumElevation.ContainsKey(i))
-					_minimumElevation[i] = GetChildAt(i).Elevation;
-				for (int j = 0; j < _minimumElevation.Count() - 1; j++)
+				if (Forms.IsLollipopOrNewer)
 				{
-					while (_minimumElevation[j] > _minimumElevation[j + 1])
+					if (!_minimumElevation.ContainsKey(i))
 					{
-						_minimumElevation[j + 1] = _minimumElevation[j] + 1;
-						GetChildAt(j + 1).Elevation = _minimumElevation[j + 1];
+						_minimumElevation[i] = GetChildAt(i).Elevation;
+					}
+						
+					for (int j = 0; j < _minimumElevation.Count() - 1; j++)
+					{
+						while (_minimumElevation[j] > _minimumElevation[j + 1])
```

Comment:
We're O(n^3) here right? And won't this condition will always be false in the case where the new elevation has the same value as a previous elevation?